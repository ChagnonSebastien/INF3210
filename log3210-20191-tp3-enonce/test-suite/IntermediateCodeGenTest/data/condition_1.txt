 Declaration
  num a;
  bool b;
 EndDeclaration

 if(false || a > 5) {
     a = 1;
     if(a < 5) {} else {
        a = a + 1;
     }
 }

 if(b) {
     a = a + 10;
 } else {
     a = a * 2;
 }

 while(a > 0) {
     a = a - 1;
 }