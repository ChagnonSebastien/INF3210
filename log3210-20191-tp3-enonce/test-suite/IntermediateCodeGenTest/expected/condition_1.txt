ifFalse a > 5 goto _L0
_L1
a = 1
ifFalse a < 5 goto _L5
goto _L6
_L5
_t2 = a + 1
a = _t2
_L6
_L0
ifFalse b == 1 goto _L10
_t3 = a + 10
a = _t3
goto _L11
_L10
_t4 = a * 2
a = _t4
_L11
_L12
ifFalse a > 0 goto _L13
_t6 = a - 1
a = _t6
goto _L12
_L13
